core/src/main/java/org/apache/brooklyn/core/entity/Entities.java

```diff
@@ -302,305 +292,190 @@ public static boolean isTrivial(Object v) {
-                else out.append(""+val.get());
-                out.append("\n");
-            }
-        }
-
-        out.flush();
+        Dumper.dumpInfo(feed, out, currentIndentation, tab);
     }
 
+    /**
+     * @deprecated since 1.0.0; instead use {@link Dumper} methods
+     */
+    @Deprecated
     public static void dumpInfo(Policy pol) {
-        try {
-            dumpInfo(pol, new PrintWriter(System.out), "", "  ");
-        } catch (IOException exc) {
-            // system.out throwing an exception is odd, so don't have IOException on signature
-            throw new RuntimeException(exc);
-        }
+        Dumper.dumpInfo(pol);
     }
+    
+    /**
+     * @deprecated since 1.0.0; instead use {@link Dumper} methods
+     */
+    @Deprecated
     public static void dumpInfo(Policy pol, Writer out) throws IOException {
-        dumpInfo(pol, out, "", "  ");
+        Dumper.dumpInfo(pol, out);
     }
+    
+    /**
+     * @deprecated since 1.0.0; such fine-grained customization will be removed from API, instead use {@link Dumper} methods
+     */
+    @Deprecated
     public static void dumpInfo(Policy pol, String currentIndentation, String tab) throws IOException {
-        dumpInfo(pol, new PrintWriter(System.out), currentIndentation, tab);
+        Dumper.dumpInfo(pol, currentIndentation, tab);
     }
+    
+    /**
+     * @deprecated since 1.0.0; such fine-grained customization will be removed from API, instead use {@link Dumper} methods
+     */
+    @Deprecated
     public static void dumpInfo(Policy pol, Writer out, String currentIndentation, String tab) throws IOException {
-        out.append(currentIndentation+pol.toString()+"\n");
-
-        for (ConfigKey<?> key : sortConfigKeys(pol.getPolicyType().getConfigKeys())) {
-            Maybe<Object> val = ((BrooklynObjectInternal)pol).config().getRaw(key);
-            if (!isTrivial(val)) {
-                out.append(currentIndentation+tab+tab+key);
-                out.append(" = ");
-                if (isSecret(key.getName())) out.append("xxxxxxxx");
-                else out.append(""+val.get());
-                out.append("\n");
-            }
-        }
-
-        out.flush();
+        Dumper.dumpInfo(pol, out, currentIndentation, tab);
     }
 
-    @SuppressWarnings({ "rawtypes", "unchecked" })
+    /**
+     * Sorts the sensors into alphabetical order by name.
+     * @deprecated since 1.0.0; viewed as unnecessary in the API, will be removed
+     */
+    @Deprecated
     public static List<Sensor<?>> sortSensors(Set<Sensor<?>> sensors) {
-        List result = new ArrayList(sensors);
-        Collections.sort(result, new Comparator<Sensor>() {
-                    @Override
-                    public int compare(Sensor arg0, Sensor arg1) {
-                        return arg0.getName().compareTo(arg1.getName());
-                    }
-
-        });
-        return result;
+        return Dumper.sortSensors(sensors);
```

Comment:
Should this be in the `Dumper` class? It does not seem to have a relation to dumping data

Reply:
It's now a package-private method of `Dumper`, used to sort the sensors when we're going to dump out the entity's details. It therefore needs to be in (or accessible from) the `Dumper` class. I don't think it should ever have been a public method (i.e. part of our sprawling api).

This is a way to deprecate it from the public api.

Reply:
Fair enough